lib/timers.js

```diff
@@ -554,56 +557,66 @@ function tryOnImmediate(immediate, queue) {
   }
 }
 
+function runCallback(timer) {
+  var argv = timer._argv;
+  var argc = argv ? argv.length : 0;
+  switch (argc) {
+    // fast-path callbacks with 0-3 arguments
+    case 0:
+      return timer._callback();
+    case 1:
+      return timer._callback(argv[0]);
+    case 2:
+      return timer._callback(argv[0], argv[1]);
+    case 3:
+      return timer._callback(argv[0], argv[1], argv[2]);
+    // more then 3 arguments run slower with .apply
+    default:
+      return timer._callback.apply(timer, argv);
+  }
+}
 
-function Immediate() { }
-
-Immediate.prototype.domain = undefined;
-Immediate.prototype._onImmediate = undefined;
-Immediate.prototype._idleNext = undefined;
-Immediate.prototype._idlePrev = undefined;
 
+function createImmediate(callback) {
+  return {
+    _idleNext: null,
+    _idlePrev: null,
+    _callback: callback,
+    _argv: null,
```

Comment:
@mscdex good tip!  It's ```[aborted optimizing 0x21a4311d <JS Function exports.setImmediate (SharedFunctionInfo 0x21a41ce5)> because: Unsupported phi use of const variable]```